edge-hub/core/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/Cloud2DeviceTest.cs

```diff
@@ -128,8 +85,16 @@ public async void Receive_C2D_SingleMessage_AfterOfflineMessage_ShouldSucceed()
                 // Dummy ReceiveAsync to ensure mqtt subscription registration before SendAsync() is called on service client.
                 await deviceClient.ReceiveAsync(TimeSpan.FromSeconds(1));
                 await deviceClient.CloseAsync();
-                // wait for the connection to be closed on the Edge side
-                await Task.Delay(TimeSpan.FromSeconds(30));
+
+                // Wait for the connection to be closed on the Edge side by checking device connection state
+                Task timerTask = Task.Delay(TimeSpan.FromMinutes(4));
```

Comment:
4 minutes? I thought we are trying to speed our tests up, not slow them down :) On a serious note though, why do we need such long timeout?

Reply:
I thought 60s would help, but it was still failing so just trying to see if 4min would help and it was still failing. I found the issue and I think with the current change is stable, still running a few more times. It needs to wait for connect before it disconnects. We are trying to fix it, not speed it :) Anyway I changed it and it waits up to 60s to connects and then disconnect and as soon this happens it can proceed, so it most cases it should be faster than 60s.